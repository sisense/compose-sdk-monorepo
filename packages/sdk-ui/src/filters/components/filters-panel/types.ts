import type { FilterEditorConfig } from '../filter-editor-popover/types';

/**
 * Configuration for the filters panel
 */
export interface FiltersPanelConfig {
  /**
   * Configures the available actions within the filters panel.
   */
  actions?: {
    /**
     * Configuration for adding a new filter.
     */
    addFilter?: {
      /**
       * Determines whether the possibility to create a new filter is enabled.
       *
       * If not specified, the default value is `false`.
       */
      enabled?: boolean;
      /** @internal */
      multiSelect?: {
        visible?: boolean;
      };
    };
    /**
     * Configuration for editing an existing filter.
     */
    editFilter?: {
      /**
       * Determines whether the possibility to edit an existing filter is enabled.
       *
       * If not specified, the default value is `false`.
       */
      enabled?: boolean;
      /** @internal */
      multiSelect?: {
        visible?: boolean;
      };
    };
    /**
     * Configuration for deleting a filter.
     */
    deleteFilter?: {
      /**
       * Determines whether the possibility to delete a filter is enabled.
       *
       * If not specified, the default value is `true`.
       */
      enabled?: boolean;
    };
  };
}

/** @internal */
export type UseExistingFilterEditingConfig = FilterEditorConfig & {
  enabled?: boolean;
};

/** @internal */
export type UseNewFilterCreationConfig = FilterEditorConfig & {
  enabled?: boolean;
};
