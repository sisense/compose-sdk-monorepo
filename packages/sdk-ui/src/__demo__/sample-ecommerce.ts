import { measureFactory } from '@sisense/sdk-data';

import { Commerce, Brand, Country, Category, DataSource } from './sample-ecommerce-autogenerated';

export { Commerce, Brand, Category, Country, DataSource };

/** adding predefined set of measures */
export const Measures = {
  SumCost: measureFactory.sum(Commerce.Cost).format('0,0$'),
  AvgCost: measureFactory.average(Commerce.Cost).format('0,0$'),
  SumRevenue: measureFactory.sum(Commerce.Revenue, 'Total Revenue').format('0,0$'),
  Quantity: measureFactory.sum(Commerce.Quantity, 'Total Quantity').format('0,0'),
  CountBrands: measureFactory.countDistinct(Commerce.BrandID).format('0,0'),
};
