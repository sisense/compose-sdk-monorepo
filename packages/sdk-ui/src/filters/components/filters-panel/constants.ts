import type { DeepRequired } from 'ts-essentials';
import { FiltersPanelConfig } from './types';

/**
 * Default configuration for the filters panel
 */
export const DEFAULT_FILTERS_PANEL_CONFIG: DeepRequired<FiltersPanelConfig> = {
  actions: {
    addFilter: {
      enabled: false,
      multiSelect: {
        visible: true,
      },
    },
    editFilter: {
      enabled: false,
      multiSelect: {
        visible: true,
      },
    },
    deleteFilter: {
      enabled: false,
    },
  },
};
