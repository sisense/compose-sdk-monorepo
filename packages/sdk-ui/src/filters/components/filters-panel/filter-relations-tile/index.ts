export * from './filter-relations-tile';
