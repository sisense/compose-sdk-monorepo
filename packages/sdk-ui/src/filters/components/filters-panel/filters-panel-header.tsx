import { useTranslation } from 'react-i18next';

import styled from '@emotion/styled';
import Button from '@mui/material/Button';

import { PlusIcon } from '@/common/icons/plus-icon';
import { useThemeContext } from '@/theme-provider';
import { Themable } from '@/theme-provider/types';

const PanelHeader = styled.div<Themable>`
  background-color: transparent;
  display: flex;
  align-items: center;
  justify-content: space-between;
  color: ${({ theme }) => theme.filter.panel.titleColor};
  margin: 0 9px;
  height: 48px;
  border-bottom: 1px solid #dadada;
  box-sizing: border-box;
`;
const PanelTitle = styled.div`
  font-size: 13px;
  font-weight: bold;
  display: flex;
  align-items: center;
  height: 100%;
  margin-left: 8px;
`;

const RoundButton = styled(Button)<Themable>`
  border-radius: 50%;
  min-width: 0;
  padding: 0;
  width: 28px;
  height: 28px;
  color: ${({ theme }) => theme.filter.panel.titleColor};

  svg path {
    fill: ${({ theme }) => theme.filter.panel.titleColor};
  }
`;

type FiltersPanelHeaderProps = {
  onAddFilterButtonClick: () => void;
  shouldShowAddFilterButton?: boolean;
};

export const FiltersPanelHeader = ({
  onAddFilterButtonClick,
  shouldShowAddFilterButton,
}: FiltersPanelHeaderProps) => {
  const { t } = useTranslation();
  const { themeSettings } = useThemeContext();

  return (
    <PanelHeader theme={themeSettings} data-testid="filters-panel-header">
      <PanelTitle>{t('filters')}</PanelTitle>
      {shouldShowAddFilterButton && <AddFilterButton onClick={onAddFilterButtonClick} />}
    </PanelHeader>
  );
};

const AddFilterButton = ({ onClick }: { onClick: () => void }) => {
  const { themeSettings } = useThemeContext();
  return (
    <RoundButton onClick={onClick} data-testid="add-filter-button" theme={themeSettings}>
      <PlusIcon />
    </RoundButton>
  );
};
