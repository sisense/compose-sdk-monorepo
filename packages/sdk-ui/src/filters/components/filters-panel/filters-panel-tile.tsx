import {
  DataSource,
  CascadingFilter,
  CustomFilter,
  DateRangeFilter,
  Filter,
  LevelAttribute,
  MeasureFilter,
  MembersFilter,
  NumericFilter,
  RankingFilter,
  RelativeDateFilter,
  TextFilter,
} from '@sisense/sdk-data';
import { MemberFilterTile } from '../member-filter-tile';
import { CriteriaFilterTile } from '../criteria-filter-tile';
import { DateRangeFilterTile, RelativeDateFilterTile } from '../date-filter';
import { FilterTile } from '../filter-tile';
import { useTranslation } from 'react-i18next';
import { CascadingFilterTile } from '../cascading-filter-tile';
import { CustomFilterTile } from '../custom-filter-tile';

/**
 * Props of the {@link FiltersPanelTile} component
 *
 * @internal
 */
export type FiltersPanelTileProps = {
  /** Filter to display */
  filter: Filter;
  /** Callback to handle filter change */
  onChange: (filter: Filter | null) => void;
  /** Default data source used for filter tiles */
  defaultDataSource?: DataSource;
};

/**
 * Filter tile component that renders a filter based on its type
 *
 * @internal
 */
export const FiltersPanelTile = ({
  filter,
  onChange,
  defaultDataSource,
}: FiltersPanelTileProps) => {
  const { t } = useTranslation();

  const attribute = filter.attribute;
  const title = attribute.name;
  const props = {
    attribute,
    title,
    onChange,
    onUpdate: onChange,
    ...(defaultDataSource ? { dataSource: defaultDataSource } : null),
  };
  // checking for custom filters first to prevent conversion attempts
  if (filter instanceof CustomFilter) {
    return <CustomFilterTile {...props} filter={filter} />;
  }

  if (filter instanceof MembersFilter) {
    return <MemberFilterTile {...props} filter={filter} />;
  }

  if (filter instanceof DateRangeFilter) {
    return (
      <DateRangeFilterTile
        {...props}
        filter={filter}
        attribute={attribute as LevelAttribute}
        tiled
      />
    );
  }

  if (filter instanceof RelativeDateFilter) {
    return <RelativeDateFilterTile {...props} filter={filter} arrangement="vertical" />;
  }

  if (
    filter instanceof MeasureFilter ||
    filter instanceof NumericFilter ||
    filter instanceof TextFilter ||
    filter instanceof RankingFilter
  ) {
    return <CriteriaFilterTile {...props} filter={filter} />;
  }

  if (filter instanceof CascadingFilter) {
    return <CascadingFilterTile filter={filter} onChange={onChange} />;
  }

  return (
    <FilterTile
      title={t('unsupportedFilter.title')}
      renderContent={() => (
        <p className="csdk-text-center csdk-text-[13px]">{t('unsupportedFilter.message')}</p>
      )}
      design={{ header: { isCollapsible: false } }}
    />
  );
};
