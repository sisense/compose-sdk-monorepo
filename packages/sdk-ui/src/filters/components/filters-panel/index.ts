export { FiltersPanel, type FiltersPanelProps } from './filters-panel';
export type { FiltersPanelConfig } from './types';
