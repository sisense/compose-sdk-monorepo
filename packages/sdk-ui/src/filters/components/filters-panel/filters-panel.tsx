import { useCallback, useRef } from 'react';
import styled from '@emotion/styled';
import { DataSource, Filter, FilterRelations, mergeFilters } from '@sisense/sdk-data';
import { FilterTile } from '@/filters/components/filter-tile';
import { Themable } from '@/theme-provider/types';
import { useThemeContext } from '@/theme-provider';
import { asSisenseComponent } from '@/decorators/component-decorators/as-sisense-component';
import { DASHBOARD_HEADER_HEIGHT } from '@/dashboard/components/dashboard-header';
import {
  calculateNewRelations,
  combineFiltersAndRelations,
  splitFiltersAndRelations,
} from '@/utils/filter-relations';
import { FilterRelationsTile } from './filter-relations-tile';
import { FiltersPanelHeader } from './filters-panel-header';
import { useNewFilterCreation } from './hooks/use-new-filter-adding';
import {
  isFilterSupportEditing,
  useExistingFilterEditing,
} from './hooks/use-existing-filter-editing';
import { FiltersPanelConfig } from './types';
import { useDefaults } from '@/common/hooks/use-defaults';
import { DEFAULT_FILTERS_PANEL_CONFIG } from './constants';

const PanelWrapper = styled.div<Themable>`
  background-color: ${({ theme }) => theme.filter.panel.backgroundColor};
  font-family: ${({ theme }) => theme.typography.fontFamily};
  border: 1px solid #dadada;
  width: fit-content;
  min-width: 238px;
  max-height: 100%;
  overflow: hidden;
`;

const PanelBody = styled.div`
  background-color: transparent;
  max-height: calc(100% - ${DASHBOARD_HEADER_HEIGHT}px);
  overflow-y: auto;
`;

const PanelBodyInner = styled.div`
  padding: 0px 12px 12px;
`;

/**
 * Props of the {@link FiltersPanel} component
 *
 */
export type FiltersPanelProps = {
  /** Array of filters to display */
  filters: Filter[] | FilterRelations;
  /** Callback to handle changes in filters */
  onFiltersChange: (filters: Filter[] | FilterRelations) => void;
  /** Default data source used for filter tiles */
  defaultDataSource?: DataSource;
  /** All data sources available for filter tiles @internal */
  dataSources?: DataSource[];
  /** The configuration for the filters panel */
  config?: FiltersPanelConfig;
};

/**
 * Filters panel component that renders a list of filter tiles
 *
 * @group Filter Tiles
 * @alpha
 */
export const FiltersPanel = asSisenseComponent({
  componentName: 'FiltersPanel',
})(
  ({
    filters: filtersOrFilterRelations,
    onFiltersChange,
    defaultDataSource,
    dataSources,
    config: propConfig,
  }: FiltersPanelProps) => {
    const filtersPanelRef = useRef<HTMLDivElement>(null);
    const filterTilesRef = useRef<HTMLElement[]>([]);
    const { themeSettings } = useThemeContext();
    const { filters, relations } = splitFiltersAndRelations(filtersOrFilterRelations);
    const config = useDefaults(propConfig, DEFAULT_FILTERS_PANEL_CONFIG);

    const handleFilterChange = useCallback(
      (changedFilter: Filter) => {
        if (!filters) return;
        const newFilters: Filter[] = filters.map((originalFilter) =>
          originalFilter.config.guid === changedFilter.config.guid ? changedFilter : originalFilter,
        );
        const newRelations = calculateNewRelations(filters, relations, newFilters);
        onFiltersChange(combineFiltersAndRelations(newFilters, newRelations));
      },
      [filters, relations, onFiltersChange],
    );

    const handleFilterAdd = useCallback(
      (newFilter: Filter) => {
        if (!filters) return;
        const newFilters = mergeFilters(filters, [newFilter]);
        const newRelations = calculateNewRelations(filters, relations, newFilters);
        onFiltersChange(combineFiltersAndRelations(newFilters, newRelations));
      },
      [filters, relations, onFiltersChange],
    );

    const handleFilterDelete = useCallback(
      (filter: Filter) => {
        if (!filters) return;
        const newFilters = filters.filter((f) => f.config.guid !== filter.config.guid);
        const newRelations = calculateNewRelations(filters, relations, newFilters);
        onFiltersChange(combineFiltersAndRelations(newFilters, newRelations));
      },
      [filters, onFiltersChange, relations],
    );

    const { ExistingFilterEditor, startEditingFilter } = useExistingFilterEditing({
      onFilterChanged: handleFilterChange,
      defaultDataSource: defaultDataSource,
      config: config.actions.editFilter,
    });

    const { NewFilterCreator, startFilterCreation } = useNewFilterCreation({
      dataSources: dataSources ? dataSources : defaultDataSource ? [defaultDataSource] : [],
      onFilterCreated: handleFilterAdd,
      defaultDataSource: defaultDataSource,
      config: config.actions.addFilter,
    });

    return (
      <PanelWrapper theme={themeSettings} ref={filtersPanelRef}>
        <FiltersPanelHeader
          onAddFilterButtonClick={useCallback(
            () => startFilterCreation(filtersPanelRef.current!),
            [startFilterCreation],
          )}
          shouldShowAddFilterButton={config.actions.addFilter.enabled}
        />
        <PanelBody>
          <PanelBodyInner>
            {relations && <FilterRelationsTile relations={relations} filters={filters} />}
            {filters?.map((filter, index) => (
              <div
                className="csdk-mt-[6px]"
                key={filter.config.guid}
                ref={(el) => (filterTilesRef.current[index] = el!)}
              >
                <FilterTile
                  onDelete={
                    config.actions.deleteFilter.enabled
                      ? () => handleFilterDelete(filter)
                      : undefined
                  }
                  key={filter.config.guid}
                  filter={filter}
                  onChange={(newFilter) => handleFilterChange(newFilter!)}
                  defaultDataSource={defaultDataSource}
                  onEdit={
                    config.actions.editFilter.enabled && isFilterSupportEditing(filter)
                      ? () => startEditingFilter(filterTilesRef.current[index], filter)
                      : undefined
                  }
                />
              </div>
            ))}
            {<ExistingFilterEditor />}
            {<NewFilterCreator />}
          </PanelBodyInner>
        </PanelBody>
      </PanelWrapper>
    );
  },
);
