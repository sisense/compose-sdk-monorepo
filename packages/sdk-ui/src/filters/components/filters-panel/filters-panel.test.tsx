import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { MockedSisenseContextProvider } from '@/__test-helpers__';
import { Attribute, Filter, filterFactory } from '@sisense/sdk-data';
import * as DM from '@/__test-helpers__/sample-ecommerce';
import { FiltersPanel } from './filters-panel';

vi.mock('@/filters/components/filter-tile', () => {
  const FilterTile = ({
    filter,
    onChange,
    onEdit,
  }: {
    filter: Filter;
    onChange: (filter: Filter) => void;
    onEdit: (filter: Filter) => void;
  }) => {
    return (
      <div data-testid="filter-tile">
        FilterTile
        {/* Button to trigger inline editing */}
        <button data-testid="inline-edit-filter-tile-button" onClick={() => onChange(filter)} />
        <button data-testid="edit-filter-button" onClick={() => onEdit(filter)} />
      </div>
    );
  };
  return { FilterTile };
});

vi.mock('@/data-browser/add-filter-popover/add-filter-data-browser', () => {
  const AddFilterDataBrowserMock = ({
    onAttributeClick,
  }: {
    onAttributeClick: (attribute: Attribute) => void;
  }) => {
    return (
      <div data-testid="add-filter-data-browser-mock">
        AddFilterDataBrowser
        <button
          data-testid="select-some-attribute-button"
          onClick={() => onAttributeClick(DM.Commerce.AgeRange)}
        >
          Select some attribute
        </button>
      </div>
    );
  };
  return { AddFilterDataBrowser: AddFilterDataBrowserMock };
});

vi.mock('@/filters/components/filter-editor-popover/filter-editor-popover', () => {
  const FilterEditorPopoverMock = ({
    filter,
    onChange,
  }: {
    filter?: Filter | null;
    onChange?: (filter: Filter) => void;
  }) => {
    return (
      <div data-testid="filter-editor-popover">
        FilterEditorPopover
        <button
          data-testid="apply-button"
          onClick={() => {
            if (filter && onChange) {
              onChange(filter);
            }
          }}
        >
          Apply
        </button>
      </div>
    );
  };
  return { FilterEditorPopover: FilterEditorPopoverMock };
});

describe('FiltersPanel', () => {
  describe('FiltersPanel â€” basic', () => {
    it('renders filters and triggers onFiltersChange', async () => {
      const filters = [
        filterFactory.members(DM.Brand.BrandID, ['1', '2']),
        filterFactory.greaterThan(DM.Commerce.Cost, 100),
      ];
      const onFiltersChange = vi.fn();

      render(
        <MockedSisenseContextProvider>
          <FiltersPanel
            filters={filters}
            onFiltersChange={onFiltersChange}
            dataSources={[DM.DataSource]}
          />
        </MockedSisenseContextProvider>,
      );

      await waitFor(() => expect(screen.getAllByTestId('filter-tile')).toHaveLength(2));
      fireEvent.click(screen.getAllByTestId('inline-edit-filter-tile-button')[0]);
      expect(onFiltersChange).toHaveBeenCalledOnce();
      expect(onFiltersChange.mock.calls[0][0]).toMatchSnapshot();
    });
  });

  describe('adding new filter', () => {
    it('opens and closes the "Add filter" popover', async () => {
      render(
        <MockedSisenseContextProvider>
          <FiltersPanel
            filters={[filterFactory.members(DM.Brand.BrandID, ['1', '2'])]}
            onFiltersChange={vi.fn()}
            dataSources={[DM.DataSource]}
            config={{
              actions: {
                addFilter: {
                  enabled: true,
                },
              },
            }}
          />
        </MockedSisenseContextProvider>,
      );

      const addFilterButton = await screen.findByTestId('add-filter-button');
      fireEvent.click(addFilterButton);

      await waitFor(() => {
        expect(screen.getByTestId('add-filter-popover')).toBeInTheDocument();
        expect(screen.getByTestId('add-filter-data-browser-mock')).toBeInTheDocument();
      });

      const popoverCloseButton = screen.getByTestId('popover-close-button');
      fireEvent.click(popoverCloseButton);
      await waitFor(() => {
        expect(screen.queryByTestId('add-filter-popover')).not.toBeInTheDocument();
      });
    });
    it('creates empty filter on attribute selection, opens filter editor, and after applying updates the filters list', async () => {
      const onFiltersChange = vi.fn();
      render(
        <MockedSisenseContextProvider>
          <FiltersPanel
            filters={[filterFactory.members(DM.Brand.BrandID, ['1', '2'])]}
            onFiltersChange={onFiltersChange}
            dataSources={[DM.DataSource]}
            config={{
              actions: {
                addFilter: {
                  enabled: true,
                },
                editFilter: {
                  enabled: true,
                },
              },
            }}
          />
        </MockedSisenseContextProvider>,
      );

      const addFilterButton = await screen.findByTestId('add-filter-button');
      fireEvent.click(addFilterButton);

      const selectAttributeButton = await screen.findByTestId('select-some-attribute-button');
      fireEvent.click(selectAttributeButton);

      await waitFor(() => {
        expect(screen.getByTestId('filter-editor-popover')).toBeInTheDocument();
      });

      const applyButton = screen.getByTestId('apply-button');
      fireEvent.click(applyButton);

      await waitFor(() => {
        expect(onFiltersChange).toHaveBeenCalledOnce();
        expect(onFiltersChange.mock.calls[0][0]).toMatchSnapshot();
      });
    });
    it('shows error when no data sources are available', async () => {
      render(
        <MockedSisenseContextProvider>
          <FiltersPanel
            filters={[filterFactory.members(DM.Brand.BrandID, ['1', '2'])]}
            onFiltersChange={vi.fn()}
            dataSources={undefined}
            config={{
              actions: {
                addFilter: {
                  enabled: true,
                },
              },
            }}
          />
        </MockedSisenseContextProvider>,
      );

      const addFilterButton = await screen.findByTestId('add-filter-button');
      fireEvent.click(addFilterButton);

      await waitFor(() => {
        expect(screen.getByLabelText('error-box')).toBeInTheDocument();
      });
    });
  });
  describe('editing existing filter', () => {
    it('opens filter editor, and after applying updates the filters list', async () => {
      const filters = [filterFactory.members(DM.Brand.BrandID, ['1', '2'])];
      const onFiltersChange = vi.fn();
      render(
        <MockedSisenseContextProvider>
          <FiltersPanel
            filters={filters}
            onFiltersChange={onFiltersChange}
            config={{
              actions: {
                editFilter: {
                  enabled: true,
                },
              },
            }}
          />
        </MockedSisenseContextProvider>,
      );

      const editFilterButton = await screen.findByTestId('edit-filter-button');
      fireEvent.click(editFilterButton);

      await waitFor(() => {
        expect(screen.getByTestId('filter-editor-popover')).toBeInTheDocument();
      });

      const applyButton = screen.getByTestId('apply-button');
      fireEvent.click(applyButton);

      await waitFor(() => {
        expect(onFiltersChange).toHaveBeenCalledOnce();
        expect(onFiltersChange.mock.calls[0][0]).toMatchSnapshot();
      });
    });
  });
});
