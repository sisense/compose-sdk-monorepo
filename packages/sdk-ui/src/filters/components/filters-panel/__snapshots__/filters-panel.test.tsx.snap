// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FiltersPanel > FiltersPanel â€” basic > renders filters and triggers onFiltersChange 1`] = `
[
  {
    "__serializable": "DimensionalElement",
    "attribute": {
      "__serializable": "DimensionalElement",
      "dataSource": {
        "live": false,
        "title": "Sample ECommerce",
      },
      "desc": "",
      "expression": "[Brand.Brand ID]",
      "name": "Brand ID",
      "type": "numeric-attribute",
    },
    "dataSource": undefined,
    "desc": "",
    "filterType": "members",
    "members": [
      "1",
      "2",
    ],
    "name": "e0b6f65ae7c",
    "type": "filter",
  },
  {
    "__serializable": "DimensionalElement",
    "attribute": {
      "__serializable": "DimensionalElement",
      "dataSource": {
        "live": false,
        "title": "Sample ECommerce",
      },
      "desc": "",
      "expression": "[Commerce.Cost]",
      "name": "Cost",
      "type": "numeric-attribute",
    },
    "dataSource": undefined,
    "desc": "",
    "filterType": "numeric",
    "name": "bf69f7fc38f",
    "operatorA": "fromNotEqual",
    "type": "filter",
    "valueA": 100,
  },
]
`;

exports[`FiltersPanel > adding new filter > creates empty filter on attribute selection, opens filter editor, and after applying updates the filters list 1`] = `
[
  {
    "__serializable": "DimensionalElement",
    "attribute": {
      "__serializable": "DimensionalElement",
      "dataSource": {
        "live": false,
        "title": "Sample ECommerce",
      },
      "desc": "",
      "expression": "[Brand.Brand ID]",
      "name": "Brand ID",
      "type": "numeric-attribute",
    },
    "dataSource": undefined,
    "desc": "",
    "filterType": "members",
    "members": [
      "1",
      "2",
    ],
    "name": "e0b6f65ae7c",
    "type": "filter",
  },
  {
    "__serializable": "DimensionalElement",
    "attribute": {
      "__serializable": "DimensionalElement",
      "dataSource": {
        "live": false,
        "title": "Sample ECommerce",
      },
      "desc": "",
      "expression": "[Commerce.Age Range]",
      "name": "Age Range",
      "type": "text-attribute",
    },
    "dataSource": undefined,
    "desc": "",
    "filterType": "members",
    "members": [],
    "name": "70e93572443",
    "type": "filter",
  },
]
`;

exports[`FiltersPanel > editing existing filter > opens filter editor, and after applying updates the filters list 1`] = `
[
  {
    "__serializable": "DimensionalElement",
    "attribute": {
      "__serializable": "DimensionalElement",
      "dataSource": {
        "live": false,
        "title": "Sample ECommerce",
      },
      "desc": "",
      "expression": "[Brand.Brand ID]",
      "name": "Brand ID",
      "type": "numeric-attribute",
    },
    "dataSource": undefined,
    "desc": "",
    "filterType": "members",
    "members": [
      "1",
      "2",
    ],
    "name": "e0b6f65ae7c",
    "type": "filter",
  },
]
`;
